Karumi/Shot
# Gradle plugin skeleton

### :pushpin: References
* **Related pull-requests:** Review #1 first please. This PR is pointing at that branch 😃 

### :tophat: What is the goal?

Create a Gradle plugin skeleton following the [official documentation](https://docs.gradle.org/current/userguide/custom_plugins.html) we will evolve in the following pull requests with all the code needed to implement the first version of our plugin.

### How is it being implemented?

The steps followed to implement this feature are:

* Update the ``shot-consumer`` project disabling the official Facebook SDK and replacing this implementation with our brand new plugin.
* Update the CI integration to don't execute screenshot tests and just build the project. We will re-enable this implementation once the plugin is finished.
* Update the project ``README.md`` file with some useful information related to how to use the project. We will update this in the future. The goal right now, is to give the PR reviewers the change to understand how the plugin will be used.
* Create three different tasks we are going to use to implement our plugin. The tasks configuration and dependencies should not change.
* Update the main plugin class to add our facebook dependency when the plugin is evaluated.

### How can it be tested?

You can install this plugin locally and use the ``shot-consumer`` module to execute the following commands:

```
./gradlew executeScreenshotTests -Precord
./gradlew executeScreenshotTests
```

You should run the android instrumentation tests and see a bunch of log traces related to the difrerent tasks execution. The order of the tasks should be the one defined in the class ``ShotPlugin``.

### Next steps?

Design the different project algebras and start working on our programs. We will start implementing the first of our tasks and validating the plugin configuration in runtime showing to the user if something went wrong with the configuration as soon as possible.
